src/test/java/com/ericsson/ei/flowtests/FlowTest.java

```diff
@@ -65,20 +70,39 @@
 
     @Mock
     private ERQueryService erQueryService;
+    
+    @Autowired
+    private RabbitTemplate rabbitMqTemplate;
 
     @Before
-    public void before() throws IOException {
-        MockitoAnnotations.initMocks(this);
-        upStreamEventsHandler.setEventRepositoryQueryService(erQueryService);
-
-        final URL upStreamResult = this.getClass().getClassLoader().getResource(UPSTREAM_RESULT_FILE);
-        ObjectMapper objectMapper = new ObjectMapper();
-        ObjectNode objectNode = objectMapper.createObjectNode();
-        objectNode.set("upstreamLinkObjects", objectMapper.readTree(upStreamResult));
-        objectNode.set("downstreamLinkObjects", objectMapper.createArrayNode());
-
-        when(erQueryService.getEventStreamDataById(anyString(), any(SearchOption.class), anyInt(), anyInt(),
-                anyBoolean())).thenReturn(new ResponseEntity<>(objectNode, HttpStatus.OK));
+    public void before() throws IOException {    	    	
+    	ObjectMapper objectMapper = new ObjectMapper();
+    	   
+    	if (!systemTest) {
+    		final URL upStreamResult = this.getClass().getClassLoader().getResource(UPSTREAM_RESULT_FILE);
+    		MockitoAnnotations.initMocks(this);
+    		upStreamEventsHandler.setEventRepositoryQueryService(erQueryService);
+    		    	
+    		ObjectNode objectNode = objectMapper.createObjectNode();
+    		JsonNode upstreamJson = objectMapper.readTree(upStreamResult);
+    		objectNode.set("upstreamLinkObjects", upstreamJson);
+    		objectNode.set("downstreamLinkObjects", objectMapper.createArrayNode());
+
+    		when(erQueryService.getEventStreamDataById(anyString(), any(SearchOption.class), anyInt(), anyInt(),
+    				anyBoolean())).thenReturn(new ResponseEntity<>(objectNode, HttpStatus.OK));
+    	} else {
+    		final URL upStreamInput = this.getClass().getClassLoader().getResource(UPSTREAM_INPUT_FILE);
+    		ArrayNode upstreamJson = (ArrayNode) objectMapper.readTree(upStreamInput);
+//    		rabbitMqTemplate.setQueue(null);
+    		if (upstreamJson != null) {
+    			for (JsonNode event : upstreamJson) {
+    				String eventStr = event.toString();
+    				rabbitMqTemplate.convertAndSend(eventStr);
+    			}    			
+    		}
+    		boolean wait = true;
```

Comment:
The boolean wait is never used.
There are commented code here that should be removed if not used.
There is a mix of tabs and spaces in this file.

Reply:
Done.